<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1574</width>
    <height>862</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="minimumSize">
       <size>
        <width>1200</width>
        <height>800</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>1</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>350</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>110</y>
         <width>141</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Рисование отрезка</string>
       </property>
      </widget>
      <widget class="QLabel" name="draw_specter_label">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>310</y>
         <width>211</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Рисование спектра отрезков</string>
       </property>
      </widget>
      <widget class="QLabel" name="analysis_label">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>580</y>
         <width>61</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Анализ:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="seg_begin_x">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>170</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>10</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="seg_begin_y">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>170</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>10</string>
       </property>
      </widget>
      <widget class="QLabel" name="begin_x_label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>170</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
      <widget class="QLabel" name="begin_y_label">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>170</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
      <widget class="QLabel" name="start_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>140</y>
         <width>111</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Начало отрезка</string>
       </property>
      </widget>
      <widget class="QLabel" name="end_x_label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>230</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="seg_end_x">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>230</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>200</string>
       </property>
      </widget>
      <widget class="QLabel" name="end_y_label">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>230</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
      <widget class="QLabel" name="end_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>200</y>
         <width>111</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Конец отрезка</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="seg_end_y">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>230</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>100</string>
       </property>
      </widget>
      <widget class="QPushButton" name="draw_seg">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>270</y>
         <width>151</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Нарисовать отрезок</string>
       </property>
      </widget>
      <widget class="QLabel" name="center_x_label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>370</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="step_degrees">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>400</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>15</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="center_x">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>370</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>600</string>
       </property>
      </widget>
      <widget class="QLabel" name="center_y_label">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>370</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
      <widget class="QPushButton" name="draw_spectre">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>460</y>
         <width>151</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Нарисовать спектр</string>
       </property>
      </widget>
      <widget class="QLabel" name="specter_step_label">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>400</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Шаг изменения угла:</string>
       </property>
      </widget>
      <widget class="QLabel" name="specter_center">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>340</y>
         <width>111</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Центр спектра</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="center_y">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>370</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>400</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="seg_length">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>430</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>500</string>
       </property>
      </widget>
      <widget class="QLabel" name="specter_length_label">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>430</y>
         <width>121</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Длина отрезков:</string>
       </property>
      </widget>
      <widget class="QPushButton" name="change_seg_color">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>500</y>
         <width>201</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Цвет отрезка</string>
       </property>
      </widget>
      <widget class="QPushButton" name="clear_screen">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>540</y>
         <width>201</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Очистить экран</string>
       </property>
      </widget>
      <widget class="QComboBox" name="algo_combobox">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>30</y>
         <width>281</width>
         <height>25</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Алгоритм ЦДА</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм Брезенхема (с действительными числами)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм Брезенхема (с целыми числами)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм Брезенхема (с устранением ступенчатости)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм Ву</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Библиотечный алгоритм</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>10</y>
         <width>301</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Выбор алгоритма построения отрезка:</string>
       </property>
      </widget>
      <widget class="QPushButton" name="analyze_steps">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>680</y>
         <width>211</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Анализ ступенчатости</string>
       </property>
      </widget>
      <widget class="QPushButton" name="analyze_speed">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>710</y>
         <width>211</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Анализ скорости работы</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="iter_amount">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>610</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>1000</string>
       </property>
      </widget>
      <widget class="QLabel" name="iter_amount_label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>610</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Количество повотрений:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="test_seg_length">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>640</y>
         <width>113</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>1000</string>
       </property>
      </widget>
      <widget class="QLabel" name="test_seg_label">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>640</y>
         <width>121</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Длина отрезка:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>301</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Координаты центра холста: X: 600; Y: 400</string>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1574</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Информация</string>
    </property>
    <addaction name="action"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
